cirq/work/observable_measurement.py

```diff
@@ -211,3 +269,148 @@ def _subdivide_meas_specs(
+            incorporated prior to measurement. In the latter case, the
+            measured bit will be flipped back classically and accumulated
+            together. This causes readout error to appear symmetric,
+            p(0|0) = p(1|1).
+        circuit_sweep: Additional parameter sweeps for parameters contained
+            in `circuit`. The total sweep is the product of the circuit sweep
+            with parameter settings for the single-qubit basis-change rotations.
+    """
+
+    qubits = _get_qubits(grouped_settings.keys())
+    qubit_to_index = {q: i for i, q in enumerate(qubits)}
+
+    needs_init_layer = False
+    for max_setting in grouped_settings.keys():
+        if _determine_needs_init_layer(max_setting.init_state):
+            needs_init_layer = True
+            break
+
+    measurement_param_circuit = _with_parameterized_layers(circuit, qubits, needs_init_layer)
+    grouped_settings = {
+        _pad_setting(max_setting, qubits): settings
+        for max_setting, settings in grouped_settings.items()
+    }
+
+    if circuit_sweep is None:
+        circuit_sweep = study.UnitSweep
+    else:
+        circuit_sweep = study.to_sweep(circuit_sweep)
+
+    # meas_spec provides a key for accumulators.
+    # meas_specs_todo is a mutable list. We will pop things from it as various
+    # specs are measured to the satisfaction of the stopping criteria
+    accumulators = {}
+    meas_specs_todo = []
+    for max_setting, circuit_params in itertools.product(
+        grouped_settings.keys(), circuit_sweep.param_tuples()
+    ):
+        # The type annotation for Param is just `Iterable`.
+        # We make sure that it's truly a tuple.
+        circuit_params = dict(circuit_params)
+
+        meas_spec = _MeasurementSpec(max_setting=max_setting, circuit_params=circuit_params)
+        accumulator = BitstringAccumulator(
+            meas_spec=meas_spec,
+            simul_settings=grouped_settings[max_setting],
+            qubit_to_index=qubit_to_index,
+        )
+        accumulators[meas_spec] = accumulator
+        meas_specs_todo += [meas_spec]
+
+    while True:
+        meas_specs_todo, repetitions = _check_meas_specs_still_todo(
+            meas_specs=meas_specs_todo,
+            accumulators=accumulators,
+            desired_repetitions=desired_repetitions,
+        )
+        if len(meas_specs_todo) == 0:
+            break
+
+        flippy_meas_specs, repetitions = _subdivide_meas_specs(
+            meas_specs=meas_specs_todo,
+            repetitions=repetitions,
+            qubits=qubits,
+            readout_symmetrization=readout_symmetrization,
+        )
+
+        resolved_params = [
+            flippy_ms.param_tuples(needs_init_layer=needs_init_layer)
+            for flippy_ms in flippy_meas_specs
+        ]
+        resolved_params = _to_sweep(resolved_params)
+        total_samples = repetitions * len(resolved_params)
+        print("Total samples", total_samples)
```

Comment:
don't we want to use some proper logging instead of print statements? In that case we can also write tests for the things we print with `cirq.testing.assert_logs`, but also timestamps, formatting, file specs, etc. can be added on top of this kind of output in general.

Reply:
Do we have an example of user-facing logging in cirq? The python `logging` module requires the user to configure something to actually get the log messages to appear and the google cloud stuff produces a lot of logging output. 

I'm removing this print statement for now. We can add better logging later.